client/src/utils/wallet.ts

```diff
@@ -0,0 +1,15 @@
+/* eslint-disable import/prefer-default-export */
+// This just disables default export as this utils file will be intended to return multiple utils
+import { getAddress } from "@ethersproject/address";
+
+export function shortAddress(address: string): string {
+  try {
+    const formattedAddress = getAddress(address);
+    return `${formattedAddress.substring(0, 6)}...${formattedAddress.substring(
+      38
+    )}`;
+  } catch (e) {
+    console.log(e, "There was an error processing your address");
+    return "Invalid Address";
+  }
+}
```

Comment:
ether's [getAddress](https://docs.ethers.io/v5/api/utils/address/#utils-getAddress) will throw an error if address is invalid

Reply:
@schultztimothy but we are not solving it with this catch right? 
cause we are just returning `"Invalid Address"` and we will send it to ethers. so in this case I would just catch the ethers error. and it theory the address will always be correct because it comes from metamask